pointpattern.py

```diff
@@ -0,0 +1,124 @@
+from point import Point
+import analytics
+import random
+import numpy as np
+import scipy.spatial as ss
+
+class PointPattern(object):
+    def __init__(self):
+        self.points = []
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def average_nearest_neighbor_distance(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, mark)
+
+    def average_nearest_neighbor_distance_kdtree(self, mark_name=None):
+        temp_points = []
+        real_points = []
+        distances = []
+
+        if mark_name is None:
+            temp_points = self.points
+        else:
+            for i, point in self.points:
+                if point.mark is mark:
+                    temp_points.append(point)
+
```

Comment:
This code looks good.  Not sure why the ipynb is throwing an error.  If you restart the kernel, is it still?